### PR TITLE
Refactor/message types to map

### PR DESCRIPTION
**Pull request by charleenfei**
A small refactor addressing #656, larger refactor of p2p handler/messagetypes to follow per #651 .

### CODE REVIEW

File: p2p/common/protocol.go [hunk 1]
```diff
@@ -51,26 +63,14 @@ func (mt MessageType) String() string {
...
-	} else if MessageTypeSendAnchoredDoc.Equals(mt) {
-		found = MessageTypeSendAnchoredDoc
-	} else if MessageTypeSendAnchoredDocRep.Equals(mt) {
-		found = MessageTypeSendAnchoredDocRep
-	} else if MessageTypeGetDoc.Equals(mt) {
-		found = MessageTypeGetDoc
-	} else if MessageTypeGetDocRep.Equals(mt) {
-		found = MessageTypeGetDocRep
+func MessageTypeFromString(ht string) MessageType {
+
```
Comment (vimukthi-git) [inline p2p/common/protocol.go]: these extra newlines are unnecessary IMO

File: p2p/common/protocol.go [hunk 2]
```diff
@@ -40,6 +40,18 @@ const (
...
+	"MessageTypeRequestSignature":    "MessageTypeRequestSignature",
+	"MessageTypeRequestSignatureRep": "MessageTypeRequestSignatureRep",
+	"MessageTypeSendAnchoredDoc":     "MessageTypeSendAnchoredDoc",
+	"MessageTypeSendAnchoredDocRep":  "MessageTypeSendAnchoredDocRep",
+	"MessageTypeGetDoc":              "MessageTypeGetDoc",
+	"MessageTypeGetDocRep":           "MessageTypeGetDocRep",
+}
+
 // Equals compares if string is of a particular MessageType
 func (mt MessageType) Equals(mt2 string) bool {
```
Comment (vimukthi-git) [inline p2p/common/protocol.go]: please remove this if not used

**Comment by vimukthi-git** [comment]
@mikiquantum I agree this is not much of an improvement in terms of adding messages, the great thing here is that adding message types is now a more declarative exercise than a logic change, I feel that to be a good thing for protocol extensibility and readability.

**Comment by mikiquantum** [comment]
@vimukthi-git couldn't we achieve both by transforming the type to int and use iota enum?

**Comment by vimukthi-git** [comment]
Sure, I think that would work as well, 
this whole discussion started because @lucasvo and I thought having `switch` case here was more idiomatic than `if-else`. But map seemed to me was even better as we can remove logic. If you have a better way to map these I'm sure I would agree as well.
 This is just the minimum change required to remove the `if-else`. IMO lets just do this now and make those changes you suggest in the future.

**Comment by charleenfei** [comment]
agree with @vimukthi-git about the temp/quick and dirty nature of this refactor. I think that it would make sense to have a larger refactor of how MessageTypes are used, including the way that HandleInterceptor routes requests to the relevant handlers, there is a lot of code redundancy there (see #651 housekeeping).

**Pull request merged**